pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -264,26 +306,97 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 	}
 
 	if shoot.Status.LastMaintenance != nil && shoot.Status.LastMaintenance.State == gardencorev1beta1.LastOperationStateProcessing {
+		maintenanceStatus := gardencorev1beta1.LastOperationStateSucceeded
 		patch := client.MergeFrom(shoot.DeepCopy())
-		shoot.Status.LastMaintenance.State = gardencorev1beta1.LastOperationStateSucceeded
+		shoot.Status.LastMaintenance.State = maintenanceStatus
```

Comment:
why do we need a new variable here?
```suggestion
		shoot.Status.LastMaintenance.State = gardencorev1beta1.LastOperationStateSucceeded
```